packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -270,17 +270,179 @@ class _UntilTextBoundary extends TextBoundary {
   }
 }
 
+class _TextLayout {
+  _TextLayout._(this._paragraph);
+
+  // This field is not final because the owner TextPainter could create a new
+  // ui.Paragraph with the exact same text layout (for example, when only the
+  // color of the text is changed).
+  ui.Paragraph _paragraph;
```

Comment:
Maybe leave a note here that whoever created the _TextLayout is in charge of disposing this _paragraph.